mate-panel/panel-background.c

```diff
@@ -1019,13 +801,7 @@ panel_background_make_string (PanelBackground *background,
 
 	effective_type = panel_background_effective_type (background);
 
-#ifdef HAVE_X11
-	if (is_using_x11 () &&
-	    (effective_type == PANEL_BACK_IMAGE ||
-	     (effective_type == PANEL_BACK_COLOR &&
-	      background->has_alpha &&
-	      !gdk_window_check_composited_wm(background->window)))) {
```

Comment:
Note that originally disabling this bit of code on Wayland was a mistake. There's no reason not to report a pixmap background on Wayland, I just got confused with all the color compositing logic.